cli/azd/cmd/up.go

```diff
@@ -90,15 +90,15 @@ func (u *upAction) Run(ctx context.Context) (*actions.ActionResult, error) {
 
 	finalOutput := []string{}
 	u.infraCreate.finalOutputRedirect = &finalOutput
-	_, err = u.infraCreate.Run(ctx)
+	_, err = actions.RunWithMiddleware(ctx, &actions.ActionOptions{Name: "infra create"}, u.infraCreate)
```

Comment:
This is an example of how the middleware chain is processed for actions run as part of a composite action.  This is an interesting one specifically because `provision` is an alias for `infra create`.  Here we are explicitly passing `infra create` as the action name and will invoke and `preinfracreate` or `postinfracreate` command hooks.